packages/react-menu/src/components/MenuItem/MenuItem.types.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { ComponentProps, ShorthandProps, ObjectSlotProp } from '@fluentui/react-utils';
+import { RefObjectFunction } from '@fluentui/react-hooks';
+
+export interface MenuItemProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Icon slot rendered before children content
+   */
+  icon?: ShorthandProps;
+}
+
+export interface MenuItemState extends MenuItemProps {
+  /**
+   * Ref to the root slot
+   */
+  ref: RefObjectFunction<HTMLElement>;
+  /**
+   * Icon slot when processed by internal state
+   */
+  icon?: ObjectSlotProp<HTMLSpanElement>; // TODO use correct props when there is a converged icon pkg
```

Comment:
```suggestion
  icon?: ObjectSlotProp<HTMLSpanElement>;
```

IMO this thing will still remain a container for icons.

Reply:
I'll remove the comment for now, but I'm not sure what the icons will look like, because mdl2 icons already have a span cointainer for example